processor/src/decorators/mod.rs

```diff
@@ -71,26 +71,26 @@ where
         let root = [self.stack.get(5), self.stack.get(4), self.stack.get(3), self.stack.get(2)];
 
         // look up the node in the advice provider
-        let node = self.advice_provider.get_tree_node(root, depth, index)?;
+        let node = self.advice_provider.get_tree_node(root, &depth, &index)?;
 
-        // write the node into the advice tape with first element written last so that it can be
+        // write the node into the advice stack with first element written last so that it can be
         // removed first
-        self.advice_provider.write_tape(AdviceSource::Value(node[3]))?;
-        self.advice_provider.write_tape(AdviceSource::Value(node[2]))?;
-        self.advice_provider.write_tape(AdviceSource::Value(node[1]))?;
-        self.advice_provider.write_tape(AdviceSource::Value(node[0]))?;
+        self.advice_provider.write_stack(AdviceSource::Value(node[3]))?;
+        self.advice_provider.write_stack(AdviceSource::Value(node[2]))?;
+        self.advice_provider.write_stack(AdviceSource::Value(node[1]))?;
+        self.advice_provider.write_stack(AdviceSource::Value(node[0]))?;
 
         Ok(())
     }
 
     /// Injects the result of u64 division (both the quotient and the remainder) at the head of
-    /// the advice tape. The stack is expected to be arranged as follows (from the top):
+    /// the advice stack. The stack is expected to be arranged as follows (from the top):
```

Comment:
Let's update this to: "Pushes the result of u64 division (both the quotient and the remainder) onto the advice stack. The operand stack is expected to be arranged as follows (from the top):".

Reply:
Done!